crates/topos-tce-storage/src/connection/handlers.rs

```diff
@@ -70,6 +72,39 @@ where
     }
 }
 
+/// Handle a GetPendingCertificates query
+///
+/// The GetPendingCertificates query will ask for pending certificate
+/// for list of subnets ids
+#[async_trait]
+impl<S> CommandHandler<GetPendingCertificates> for Connection<S>
+where
+    S: Storage,
+{
+    type Error = StorageError;
+
+    /// Handle a GetPendingCertificates query
+    /// Return vector of tuples (PendingCertificateId, pending Certificate) sorted by PendingCertificateId
+    async fn handle(
+        &mut self,
+        _: GetPendingCertificates,
+    ) -> Result<Vec<(PendingCertificateId, Certificate)>, StorageError> {
+        let mut pending_certificates =
+            self.storage.get_pending_certificates().await.map_err(|e| {
+                error!("Failure on the storage to get pending certificates: {e}");
+                e
+            })?;
+
+        pending_certificates.sort_by(|(pending_cert_id_a, _), (pending_cert_id_b, _)| {
+            pending_cert_id_a
+                .partial_cmp(pending_cert_id_b)
+                .unwrap_or(Ordering::Equal)
+        });
```

Comment:
I don't think you need to sort them, they are already sorted, I guess.

Reply:
They are, but just in case and no harm is done

Reply:
It just adds runtime overhead to go through the whole list

Reply:
It adds microsecond or two to couple of hundred of milliseconds http call

Reply:
As we will refactor handling of the pending certificates anyway better for sorting to stay here to communicate what we need